conda/models/environment.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""Conda environment data model"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from logging import getLogger
+from typing import TYPE_CHECKING
+
+from ..exceptions import CondaValueError
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from .match_spec import MatchSpec
+
+
+log = getLogger(__name__)
+
+
+@dataclass
+class Environment:
+    # Prefix the environment is installed into (required).
+    prefix: str
+
+    # The platform this environment may be installed on (required)
+    platform: str
+
+    # Environment level configuration, eg. channels, solver options, etc.
+    config: dict[str, Any] = field(default_factory=dict)
+
+    # Map of other package types that conda can install. For example pypi packages.
+    external_packages: dict[str, list] = field(default_factory=dict)
```

Comment:
```suggestion
    external_packages: dict[str, list[Any] = field(default_factory=dict)
```

Reply:
hmmmm, intially I was imaging this as a `list[str]`. But, I can see where we may want to have a more specific datatype. Taking this approach will have some implications on how environments get merged. Thanks for raising this. Will look into it!